### PR TITLE
Add support for integration type

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Allow configuring integrations marked as "helper" from the helpers page.

image

image

To do:
- [x] Allow deletion

Requires https://github.com/home-assistant/core/pull/68349

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/helpers/ha-config-helpers.ts [hunk 1]
```diff
@@ -37,109 +69,133 @@ export class HaConfigHelpers extends SubscribeMixin(LitElement) {
 
   @state() private _stateItems: HassEntity[] = [];
 
-  @state() private _entitiesConfigEntryLookup?: Record<string, string>;
+  @state() private _entityEntries?: Record<string, EntityRegistryEntry>;
 
-  @state() private _configEntries?: ConfigEntry[];
+  // @state() private _entitiesConfigEntryLookup?: Record<string, string>;
```
Comment (zsarnett) [inline src/panels/config/helpers/ha-config-helpers.ts:74]:
```suggestion
```

File: src/panels/config/helpers/ha-config-helpers.ts [hunk 2]
```diff
@@ -33,98 +63,124 @@ export class HaConfigHelpers extends LitElement {
 
   @state() private _stateItems: HassEntity[] = [];
 
-  private _columns = memoize((narrow, _language): DataTableColumnContainer => {
-    const columns: DataTableColumnContainer = {
-      icon: {
+  @state() private _entityEntries?: Record<string, EntityRegistryEntry>;
+
+  // @state() private _entitiesConfigEntryLookup?: Record<string, string>;
```
Comment (zsarnett) [inline src/panels/config/helpers/ha-config-helpers.ts:68]:
```suggestion
```

File: src/translations/en.json [hunk 3]
```diff
@@ -810,7 +810,8 @@
           "area": "Set entity area only",
           "area_note": "By default the entities of a device are in the same area as the device. If you change the area of this entity, it will no longer follow the area of the device.",
           "follow_device_area": "Follow device area",
-          "change_device_area": "Change device area"
+          "change_device_area": "Change device area",
+          "configure_state": "Configure State"
```
Comment (zsarnett) [inline src/translations/en.json:814]: Edit Options?

I think the word `edit` is better than configure for these. State is weird as you aren't changing the state. you could change the options of a select. which doesn't change the state

Comment (zsarnett) [reply src/translations/en.json:814]: Right. So configure state doesn't make sense for everything. The dialog is also called options.

Comment (balloob) [reply src/translations/en.json:814]: The dialog is called settings:

image

Comment (zsarnett) [reply src/translations/en.json:814]: The dialog that configure state opens is that called options like it was in your edemo the other day?

**Review by zsarnett — APPROVED** [review]

**Pull request merged**